src/Cli.Tests/ConfigureOptionsTests.cs

```diff
@@ -150,5 +150,211 @@ public void TestUpdateDepthLimitForGraphQL(int? newDepthLimit)
 
             Assert.AreEqual(newDepthLimit, config.Runtime?.GraphQL?.DepthLimit);
         }
+
+        /// <summary>
+        /// Tests the update of the database type in the runtime config.
+        /// dab configure `--data-source.database-type {dbType}`
+        /// This method verifies that the database type can be updated to various valid values, including different cases,
+        /// and ensures that the config file is correctly modified and parsed after the update.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow("mssql", DisplayName = "Update the database type to MSSQL")]
+        [DataRow("MSSql", DisplayName = "Update the database type to MSSQL with different case")]
+        [DataRow("postgresql", DisplayName = "Update the database type to PostgreSQL")]
+        [DataRow("cosmosdb_nosql", DisplayName = "Update the database type to CosmosDB_NoSQL")]
+        [DataRow("cosmosdb_postgresql", DisplayName = "Update the database type to CosmosDB_PGSQL")]
+        [DataRow("mysql", DisplayName = "Update the database type to MySQL")]
+        public void TestDatabaseTypeUpdate(string dbType)
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_CONFIG));
+
+            Assert.IsTrue(_fileSystem!.File.Exists(TEST_RUNTIME_CONFIG_FILE));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config.Runtime);
+
+            ConfigureOptions options = new(
+                dataSourceDatabaseType: dbType,
+                config: TEST_RUNTIME_CONFIG_FILE
+            );
+
+            // Act
+            bool isSuccess = TryConfigureSettings(options, _runtimeConfigLoader!, _fileSystem!);
+
+            // Assert
+            Assert.IsTrue(isSuccess);
+            string updatedConfig = _fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config.Runtime);
+            Assert.AreEqual(config.DataSource.DatabaseType, Enum.Parse<DatabaseType>(dbType, ignoreCase: true));
+        }
+
+        /// <summary>
+        /// Tests the update of the database type from CosmosDB_NoSQL to MSSQL in the runtime config.
+        /// This method verifies that the database type can be changed from CosmosDB_NoSQL to MSSQL and that the 
+        /// specific MSSQL option 'set-session-context' is correctly added to the configuration and the specific
+        /// cosmosDB options are removed.
+        /// Command: dab configure --data-source.database-type mssql --data-source.options.set-session-context true 
+        /// </summary>
+        [TestMethod]
+        public void TestDatabaseTypeUpdateCosmosDB_NoSQLToMSSQL()
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_COSMOSDB_NOSQL_CONFIG));
```

Comment:
Looks like shared code between the //arrange steps in these tests, could refactor this out to eliminate the duplication.